src/main/src/main/java/org/geoserver/catalog/impl/DimensionInfoImpl.java

```diff
@@ -143,51 +170,26 @@ public int hashCode() {
         result = prime * result + ((unitSymbol == null) ? 0 : unitSymbol.hashCode());
         result = prime * result + ((presentation == null) ? 0 : presentation.hashCode());
         result = prime * result + ((resolution == null) ? 0 : resolution.hashCode());
+        result = prime * result + ((nearestMatchEnabled == null) ? 0 : resolution.hashCode());
+        result = prime * result + ((acceptableInterval == null) ? 0 : resolution.hashCode());
         return result;
     }
 
     @Override
-    public boolean equals(Object obj) {
-        if (this == obj)
-            return true;
-        if (obj == null)
-            return false;
-        if (getClass() != obj.getClass())
-            return false;
-        DimensionInfoImpl other = (DimensionInfoImpl) obj;
-        if (attribute == null) {
-            if (other.attribute != null)
-                return false;
-        } else if (!attribute.equals(other.attribute))
-            return false;
-        if (units == null) {
-            if (other.units != null)
-                return false;
-        } else if (!units.equals(other.units))
-            return false;
-        if (unitSymbol == null) {
-            if (other.unitSymbol != null)
-                return false;
-        } else if (!unitSymbol.equals(other.unitSymbol))
-            return false;
-        if (enabled != other.enabled)
-            return false;
-        if (presentation == null) {
-            if (other.presentation != null)
-                return false;
-        } else if (!presentation.equals(other.presentation))
-            return false;
-        if (resolution == null) {
-            if (other.resolution != null)
-                return false;
-        } else if (!resolution.equals(other.resolution))
-            return false;
-        if (endAttribute == null) {
-            if (other.endAttribute != null)
-                return false;
-        } else if (!endAttribute.equals(other.endAttribute))
-            return false;
-        return true;
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        DimensionInfoImpl that = (DimensionInfoImpl) o;
+        return enabled == that.enabled &&
+                Objects.equals(attribute, that.attribute) &&
+                Objects.equals(endAttribute, that.endAttribute) &&
+                presentation == that.presentation &&
+                Objects.equals(resolution, that.resolution) &&
+                Objects.equals(units, that.units) &&
+                Objects.equals(unitSymbol, that.unitSymbol) &&
+                Objects.equals(defaultValue, that.defaultValue) &&
+                Objects.equals(nearestMatchEnabled, that.nearestMatchEnabled) &&
+                Objects.equals(acceptableInterval, that.acceptableInterval);
```

Comment:
Thank you ! Now we can actually understand what is going on :yum:

Reply:
Courtesy of IntelliJ equals generator :-p